### ISSUE TITLE
Not possible to toggle audio filters in menu

### ISSUE THREAD

**Issue by runeflobakk** [iina/iina#3462]
**System and IINA version:**

- macOS Catalina 10.15.7
- IINA 1.1.2 (and also all earlier versions I can remember)

**Expected behavior:**

Clicking menu item _Audio -> Saved Audio Filters_ should enable me to click any of my saved audio filters to toggle them on or off.

Suggestion: clicking an audio filter here should not close the menu but add an indicator of the filter being enabled or not. This will enable adding multiple filters without having to navigate to the menu for each filter.

**Actual behavior:**

I see my saved audio filters, but each item is "disabled" (grey) and not clickable.

Screenshot:
<img width="643" alt="saved-audio-filters-not-clickable" src="https://user-images.githubusercontent.com/174823/127378995-c6f5e12d-3a7e-4d2e-a1ab-84f848212e66.png">

**Steps to reproduce:**

- Add an audio filter, e.g: Custom (FFmpeg) - Filter value: `pitch=1.05946`, Filter name: `rubberband`
- click "Save" button next to the added filter, name it something eg "Pitch +1"
- disable it by clicking "-"
  - verify the saved filter is not clickable in _Audio -> Saved Audio Filters_
- click the checkbox next to the saved filter in the _Audio Filters..._ dialog to enable the filter
  - verify the saved filter now has a ✔ next to it in _Audio -> Saved Audio Filters_

**How often does this happen?**

I don't think I have seen the saved filters being clickable in earlier versions either. I always have to open the _Audio Filters..._ window to toggle the saved filters.

Thanks for making IINA! 🏆

**Comment by runeflobakk** [comment]
I just noticed that the menu works as intended when playing video files. I only use IINA for audio files, so for some reason the menu items are disabled when playing audio files specifically.

**Comment by low-batt** [comment]
See the analysis of filter problems discussed in issue #3620. Filters that support multiple parameters have more than one valid string representation. Filters must be compared using their object representation. Code needed to be changed in `MenuController` and `MainWindowMenuActions`. This problem is more likely to be seen with audio filters as the ones frequently used take multiple parameters.

**Comment by low-batt** [comment]
The fix in PR #3665 has been merged into the develop branch.

**Comment by low-batt** [comment]
Closing. Fixed in IINA 1.3.0.

**Comment by runeflobakk** [comment]
I just tried to upgrade to 1.3.0, and the issue seems to still be present. As far as I can see, the behavior is the same: the audio filters can indeed be toggled in the menu when playing _video_, but not when playing audio files (mp3, etc).

**Comment by low-batt** [comment]
@runeflobakk Sorry! I found it was not working when playing video files for certain audio filters. I failed to notice this about it not working when playing audio files:
>I just noticed that the menu works as intended when playing video files. I only use IINA for audio files, so for some reason the menu items are disabled when playing audio files specifically.

With 1.3.0 I am seeing the same behavior of the filters being inaccessible when playing an audio file. I will investigate and open a new issue against 1.3.0. With all the many changes in 1.3.0, along with major upgrades to the libraries from other projects we were expecting to have to follow up with maintenance release to correct regressions. I will see if we can get this fixed for that release.

Thanks so much for testing 1.3.0 and reporting this is not fixed.
